Tests/SwiftFormatTests/Core/IgnoreFileTests.swift

```diff
@@ -0,0 +1,126 @@
+@_spi(Internal) import SwiftFormat
+import XCTest
+
+final class IgnoreFileTests: XCTestCase {
+  var testTreeURL: URL?
+
+  /// Description of a file or directory tree to create for testing.
+  enum TestTree {
+    case file(String, String)
+    case directory(String, [TestTree])
+  }
+
+  override func tearDown() {
+    // Clean up any test tree after each test.
+    if let testTreeURL {
+      // try? FileManager.default.removeItem(at: testTreeURL)
+    }
+  }
+
+  /// Make a temporary directory tree for testing.
+  /// Returns the URL of the root directory.
+  /// The tree will be cleaned up after the test.
+  /// If a tree is already set up, it will be cleaned up first.
+  func makeTempTree(_ tree: TestTree) throws -> URL {
+    if let testTreeURL {
+      try? FileManager.default.removeItem(at: testTreeURL)
+    }
+    let tempDir = FileManager.default.temporaryDirectory
+    let tempURL = tempDir.appendingPathComponent(UUID().uuidString)
+    try FileManager.default.createDirectory(at: tempURL, withIntermediateDirectories: true)
+    try writeTree(tree, to: tempURL)
+    testTreeURL = tempURL
+    return tempURL
+  }
```

Comment:
What do you think about making this a `withTempTree` function that deletes the temporary directory after it executes the tests? Otherwise these trees will be left as artifacts. Ie something like

```suggestion
  private func withTempTree(_ tree: TestTree, _ body: (URL) throws -> Void) throws {
    let tempURL = FileManager.default.temporaryDirectory.appendingPathComponent("swift-format-test-scratch").appendingPathComponent(UUID().uuidString)
    defer {
      try? FileManager.default.removeItem(at: tempURL)
    }
    try FileManager.default.createDirectory(at: tempURL, withIntermediateDirectories: true) // I would move this to `writeTree`
    try writeTree(tree, to: tempURL)
    body(tempURL)
  }
```